sandy/sections/mf9.py

```diff
@@ -0,0 +1,154 @@
+
+Function `write_mf9` writes a content object for a MF9/MT section into a
+string.
+MAT, MF, MT and line numbers are also added (each line ends with a `\n`).
+"""
+__author__ = "Aitor Bengoechea"
+__all__ = [
+        "read_mf9",
+        "write_mf9",
+        ]
+
+import sandy
+
+
+def read_mf9(tape, mat, mt):
+    """
+    Parse MAT/MF=9/MT section from `sandy.Endf6` object and return
+    structured content in nested dcitionaries.
+
+    Parameters
+    ----------
+    tape : `sandy.Endf6`
+        endf6 object containing requested section
+    mat : `int`
+        MAT number
+    mt : `int`
+        MT number
+
+    Returns
+    -------
+    `dict`
+        Content of the ENDF-6 tape structured as nested `dict`.
+
+    Examples
+    --------
+    Endf-6 tape structured 'dict' of Radiactive capture of Am-241 from
+    the ENDFB-VII.1 library to obtain the multiplicities for production
+    of radioactive nuclides
+    >>> read_mf9(sandy.get_endf6_file("endfb_71", 'xs', 952410),9543,102)
+    {'MAT': 9543,
+     'MF': 9,
+     'MT': 102,
+     'ZA': 95241.0,
+     'AWR': 238.986,
+     'LIS': 0,
+     'NS': 2,
+     'subsection1': {'QM': 5539101.0,
+                     'QI': 5539101.0,
+                     'IZAP': 95242,
+                     'LFS': 0,
+                     'NBT': [9],
+                     'INT': [3],
+                     'E': array([1.000000e-05, 3.690000e-01, 1.000000e+03, 1.000000e+05,
+                                 6.000001e+05, 1.000000e+06, 2.000000e+06, 4.000001e+06,
+                                 3.000000e+07]),
+                     'Y': array([0.9    , 0.9    , 0.8667 , 0.842  , 0.81533, 0.74382, 0.5703 ,
+                                 0.52   , 0.52   ])},
+     'subsection2': {'QM': 5539101.0,
+                     'QI': 5490471.0,
+                     'IZAP': 95242,
+                     'LFS': 2,
+                     'NBT': [9],
+                     'INT': [3],
+                     'E': array([1.000000e-05, 3.690000e-01, 1.000000e+03, 1.000000e+05,
+                                 6.000001e+05, 1.000000e+06, 2.000000e+06, 4.000001e+06,
+                                 3.000000e+07]),
+                     'Y': array([0.1    , 0.1    , 0.1333 , 0.158  , 0.18467, 0.25618, 0.4297 ,
+                                 0.48   , 0.48   ])}}
+    """
+    mf = 9
+    df = tape._get_section_df(mat, mf, mt)
+    out = {"MAT": mat, "MF": mf, "MT": mt}
+    i = 0
+    C, i = sandy.read_cont(df, i)
+    out.update({
+        "ZA": C.C1,
+        "AWR": C.C2,
+        "LIS": C.L1,
+        "NS": C.N1,
+        })
+    for hx in range(C.N1):
+        T, i = sandy.read_tab1(df, i)
+        add = {}
+        add = {
+                "QM": T.C1,
+                "QI": T.C2,
+                "IZAP": T.L1,
+                "LFS": T.L2,
+                "NBT": T.NBT,
+                "INT": T.INT,
+                "E": T.x,
+                "Y": T.y,
+              }
+        out["subsection" + str(hx+1)] = add
```

Comment:
Use a different key other than "`subsection" + str(hx+1)`. The `"LFS"` number should be unique, then I would use that as key.

Also, these subsections should be on a deeper dictionary level 
```
    subsections = {}
    for hx in range(C.N1):
        T, i = sandy.read_tab1(df, i)
        LFS = T.L2
        add = {
                "QM": T.C1,
                "QI": T.C2,
                "IZAP": T.L1,
                "NBT": T.NBT,
                "INT": T.INT,
                "E": T.x,
                "Y": T.y,
              }
        subsections[LFS] = add
    out["LFS"] = subsections
```